jwt/main.go

```diff
@@ -13,35 +14,32 @@ func handle_auth(w rest.ResponseWriter, r *rest.Request) {
 }
 
 func main() {
-	jwt_middleware := jwt.JWTMiddleware{
+	jwt_middleware := &jwt.JWTMiddleware{
 		Key:        []byte("secret key"),
 		Realm:      "jwt auth",
 		Timeout:    time.Hour,
 		MaxRefresh: time.Hour * 24,
 		Authenticator: func(userId string, password string) bool {
-			if userId == "admin" && password == "admin" {
-				return true
-			}
-			return false
+			return userId == "admin" && password == "admin"
 		}}
 
-	login_api := rest.NewApi()
-	login_api.Use(rest.DefaultDevStack...)
-	login_router, _ := rest.MakeRouter(
+	api := rest.NewApi()
+	// we use the IfMiddleware to remove certain paths from needing authentication
+	api.Use(&rest.IfMiddleware{
+		Condition: func(request *rest.Request) bool {
+			return request.URL.Path == "/login"
+		},
+		IfTrue:  nil,
+		IfFalse: jwt_middleware,
+	})
+	api.Use(rest.DefaultDevStack...)
```

Comment:
The DefaultDevStack includes error handling and logging. I think it should be moved before `api.Use(&rest.IfMiddleware{`.
That was, auth error will be correctly logged.